src/validator.js

```diff
@@ -0,0 +1,52 @@
+const { get, isNil } = require("lodash");
+const buildRoutingGraph = require("../src/helpers/build-routing-graph");
+const detectOrphan = require("./features/routing/detect-orphan");
+
+const failFast = errors => {
+  if (errors.errors.length > 0) {
+    throw new Error(JSON.stringify(errors, null, 2));
+  }
+};
+
+//eslint-disable-next-line
+const validatePage = (page, ctx, errors, options) => {
+  //Call your page-level validation scripts here
+  const graph = buildRoutingGraph(ctx);
+  const orphanError = detectOrphan(page.id, ctx, graph);
+  if (!isNil(orphanError)) errors.errors.push(orphanError); //eslint-disable-line
+
+  return errors;
+};
+
+//eslint-disable-next-line
+const validateSection = (section, ctx, errors, options) => {
+  //Call your section-level validation scripts here
+  return errors;
+};
+
+//eslint-disable-next-line
+const validateSurvey = (ctx, errors, options) => {
+  //Call your survey level validation scripts here
+  return errors;
+};
+
+const validateEverything = (ctx, errors, options) => {
```

Comment:
Can we default these error objects just so people aren't forced into providing their own?

Reply:
For sure. I'll do that.

Reply:
```
it("Wont throw an error if there aren't any", () => {
    const errors = {
      errors: []
    };
    expect(() => validateEverything(validTestData, errors)).not.toThrow();
  });
  it("Will throw an error if there is one", () => {
    expect(() => validateEverything(orphanedPageTestData)).toThrow();
  });
```

In `validator.test.js` I've had one of the tests use a defined errors object, and one not, to demonstrate that it works both ways.